files/__addonLocation__/package.json

```diff
@@ -102,8 +102,8 @@
       "default": "./dist/index.js"
     },
     "./*": {
-      "types": "./declarations/*.d.ts",
-      "default": "./dist/*.js"
+      "types": ["./declarations/*.d.ts", "./declarations/*/index.d.ts"],
+      "default": ["./dist/*.js", "./dist/*/index.js"]
     },<% } else { %>
     ".": "./dist/index.js",
     "./*": "./dist/*.js",<% } %>
```

Comment:
does this syntax support the array form?

Reply:
yeah, like, if we add `/index` stuff, it should be both for and without the TS exports

Reply:
Ah right! Not sure why I didn't even think about it. I'll add it, _should_ we agree to actually go forward with this!

Reply:
I mean, you have my vote!

Reply:
But what about vite? My understanding is this will fail under vite. Unless we can convince vite maintainers to change course? (not too likely?)

Reply:
Why's vite matter here?

The need to declare something about index for ember's conventions helps all packagers